explorer/interpreter/value.cpp

```diff
@@ -33,6 +36,18 @@ auto StructValue::FindField(std::string_view name) const
   return std::nullopt;
 }
 
+static auto FindClassField(Nonnull<const NominalClassValue*> object,
```

Comment:
Just a note, no change requested:

I think this approach is fine for now, but in the longer term, the type checker should be inserting conversions where necessary to convert derived class expressions to base class expressions so that the interpreter only ever needs to look for a field in the current class, rather than searching its base classes. (Once we support explicitly disambiguating a lookup with `object.(Derived.Field)` vs `object.(Base.Field)`, I think we'll need this.)

Reply:
Agree, I didn't see a way to leverage / test that today, without explicit disambiguation, and kept that aside for now.